### PR TITLE
#645 Part 5: Incorporate pathology feedback

### LINKED ISSUE #645: Proposed Analysis: Include pathology feedback in repository
Related to #609 

#### What are the goals of the analysis?

In https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/609#issuecomment-602821376, we have some feedback from CHOP pathologists Maria Rita Santi and Angela Viaene on `integrated_diagnosis`.  Many of the `Notes` contain:

> Updated via OpenPBTA subtyping

Which is to say are the results of molecular subtyping modules included in the repository.  From that same comment: 

> 1. `PT_7E3V3JFX` specimens were consistent with the original EPN dx, so pathology would call this a rare EPN, H3 K28 mutated tumor, rather than DMG.
> 2. `PT_AQWDQW27` specimen was consistent with meningioma, even though it has a hallmark EPN fusion, so pathology would also call this a rare meningioma with a _YAP1_ fusion.
> 3. Because 1 is a rare tumor (maybe first seen), the logic of searching for all H3 K28 mutations in [HGG subtyping](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/249) would convert this sample by default - how to handle this? 
> 4. Pathology confirmed this HGG BRAF V600E mutant tumor, `BS_H1XPVS9A` [https://cbethell.github.io/open-pbta-output/09-HGG-with-braf-clustering.nb.html#identify_sample_that_clusters_with_lgat), to be a LGAT (PXA). I updated `molecular_subtype` here based on what it would look like, but this should come through via the LGAT [subtyping](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/631) ticket. How should we add this info?

I think a short-term solution to address all of these notes in this repository is to create a new module `molecular-subtyping-pathology` that is essentially downstream of all other molecular subtyping modules that:

1. Compiles all the subtyping calls from all modules
2. Addresses these notes in a notebook that documents why changes need to be made

_Note: now might be a good time to consider grouping all of the molecular subtyping modules a directory (`analyses/molecular-subtyping`) where that directory has a README that explains the need for `molecular-subtyping-pathology` - alternatively put the downstream steps in `analyses/molecular-subtyping` itself. Might also consider a refactor to include common helper functions but that's a future problem._

#### How long do you expect is needed to complete the analysis? Will it be a multi-step analysis?

~1 week - it will be multistep if we address the italicized note above

#### Who will complete the analysis (please add a GitHub handle here if relevant)?

@jaclyn-taroni

### PR DESCRIPTION
**Pull request by jaclyn-taroni**
⚠️ **Depends on #662**

### Purpose/implementation Section

Copying this over from #658 which we are breaking up into more focused pull requests.

> We have multiple molecular subtyping modules that output a table of molecular subtype labels or result in the reclassification of samples based on the logic that is laid out in the repository. We received feedback from CHOP pathologists Maria Rita Santi and Angela Viaene on https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/609#issuecomment-602821376. 
> 
> The rationale for #645, which this pull request addresses, is to make sure that any updates that are either the result of molecular subtyping or deviate from the logic in a molecular subtyping are captured somewhere in the repository downstream of the other molecular subtyping modules – specifically:
> 
>    * [`molecular-subtyping-EWS`](https://github.com/jaclyn-taroni/OpenPBTA-analysis/tree/645-pathology-feedback/analyses/molecular-subtyping-EWS)
>    * [`molecular-subtyping-HGG`](https://github.com/jaclyn-taroni/OpenPBTA-analysis/tree/645-pathology-feedback/analyses/molecular-subtyping-HGG)
>    * [`molecular-subtyping-LGAT`](https://github.com/jaclyn-taroni/OpenPBTA-analysis/tree/645-pathology-feedback/analyses/molecular-subtyping-LGAT)
>    * [`molecular-subtyping-embryonal`](https://github.com/jaclyn-taroni/OpenPBTA-analysis/tree/645-pathology-feedback/analyses/molecular-subtyping-embryonal)

In essence, I think it's beneficial to have this information under version control in the code base itself rather than only in a comment on an issue.

#### What was your approach?

For each of the notes stated on https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/609#issuecomment-602821376 and the HGG BRAF V600E point in https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/627#issuecomment-598789232, there is a section that updates the compiled molecular subtyping results and adds some documentation around those changes.

#### What GitHub issue does your pull request address?

Closes #645 

### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

This includes changes from #662 at the moment, the main change that needs to get looked at here is `analyses/molecular-subtyping-pathology/02-incorporate-pathology-feedback.Rmd` and the output `analyses/molecular-subtyping-pathology/results/compiled_molecular_subtypes_with_pathology_feedback.tsv`.

#### Which areas should receive a particularly close look? /  Is there anything that you want to discuss further?

Does this notebook meet the stated goal of recording the feedback from pathology in the repository?

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/molecular-subtyping-pathology/02-incorporate-pathology-feedback.Rmd [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    broad_histology = case_when(
+      Kids_First_Participant_ID == "PT_7E3V3JFX" ~ "Ependymal tumor",
+      TRUE ~ broad_histology
+    ),
+    molecular_subtype = case_when(
+      Kids_First_Participant_ID == "PT_7E3V3JFX" ~ "EPN, H3 K28",
+      TRUE ~ molecular_subtype
+    )
+  )
+```
```
Comment (kgaonkar6) [inline analyses/molecular-subtyping-pathology/02-incorporate-pathology-feedback.Rmd:160]: Like you've checked for BRAF fusion in the newly changed HGG => LGG,BRAF V600E subtype below,  do you think we need to check for any other subtyping ( I think they might be subtyped as RELA/YAP1 fusion ) in ependymoma along with the EPN ,H3 K28M ?

Comment (jaclyn-taroni) [reply analyses/molecular-subtyping-pathology/02-incorporate-pathology-feedback.Rmd:160]: The ependymoma table hasn't been completed yet (see open PR #555), so there's no EPN table included in `01-compile-subtyping-results.Rmd`. So I think it makes sense to tackle this suggestion at the same time as adding EPN into `molecular-subtyping-pathology` more generally.

Comment (jaclyn-taroni) [reply analyses/molecular-subtyping-pathology/02-incorporate-pathology-feedback.Rmd:160]: Once #645 gets closed, I will file an update an analysis ticket about all the EPN todos.

Comment (kgaonkar6) [reply analyses/molecular-subtyping-pathology/02-incorporate-pathology-feedback.Rmd:160]: ok sounds good!

**Comment by jaclyn-taroni** [comment]
Thanks for the review @kgaonkar6 !

**Pull request merged**